test/server.coffee

```diff
@@ -34,24 +34,25 @@ describe 'Server', ->
       it 'should keep the connection active', ->
         client = sugar.createClient 'user_id=1&auth_token=valid_auth'
 
-        client.ping().then ->
-          expect(client.keepAliveTimer).to.change.when(client.ping)
+        client.pong().then ->
+          expect(client.keepAliveTimer).to.change.when(client.pong)
 
       it 'should mark the user as active', ->
         client = sugar.createClient 'user_id=1&auth_token=valid_auth'
         sugar.presence.userActiveOn = chai.spy sugar.presence.userActiveOn
 
         client.once 'connected', ->
           client.subscribeTo('zooniverse').then ->
-            client.ping().then ->
+            client.pong().then ->
               expect(sugar.presence.userActiveOn).to.have.been.called.with('zooniverse', 'user:1').at.least.once
 
   describe 'disconnecting', ->
     it 'should clear the keepAliveTimer', ->
       client = sugar.createClient()
       client.spark().then (spark) ->
-        client.ping().then ->
-          expect(spark.keepAliveTimer).to.not.eql null
+        expect(spark.keepAliveTimer).to.eql null
+        client.pong().then ->
+          # expect(spark.keepAliveTimer).to.not.eql null
```

Comment:
I think this test was making sure the client connections are cleaned up server side if the client doesn't respond within the window - so i think we need this functionality as it' used for user presence detection.

Specifically i think our server need to have the ` spark.on 'incoming::ping', (->` function modified to meet the new inverted protocol, `spark.on 'incoming::pong', (->

https://github.com/zooniverse/sugar/blob/e9151de47bf3ac98fe95d0be12c681dcc5bfc9e0/lib/server.coffee#L150-L151

Reply:
Thanks. I haven’t really dived into the server code. Now that the ping/pong protocol is inverted, should this test run `server.spark()` and test the keepAlive timer on that?

Reply:
`client.pong()` is probably wrong here. If the server sends a ping, there’ll be no answering pong from a disconnected client.

Reply:
Updating the server to use `incoming::pong` fixes this test. 🤔